### ISSUE TITLE
IBM tool/init_after_finalize failures

### ISSUE THREAD

**Issue by hppritcha** [open-mpi/ompi#9013]
I am seeing asserts with the IBM tool/init_after_finalize test of the pattern
```
closing opal framework btl  <------------ my debug statement
init_after_finalize: mca_base_var_group.c:375: mca_base_var_group_deregister: Assertion `((0xdeafbeedULL << 32) + 0xdeafbeedULL) == ((opal_object_t *) (enums[i]))->obj_magic_id' failed.
[st-master:28398] *** Process received signal ***
[st-master:28398] Signal: Aborted (6)
[st-master:28398] Signal code:  (-6)
[st-master:28398] [ 0] [0xffffbd43066c]
[st-master:28398] [ 1] /lib64/libc.so.6(gsignal+0x4c)[0xffffbcd950e8]
[st-master:28398] [ 2] /lib64/libc.so.6(abort+0x11c)[0xffffbcd96760]
[st-master:28398] [ 3] /lib64/libc.so.6(+0x2e5dc)[0xffffbcd8e5dc]
[st-master:28398] [ 4] /lib64/libc.so.6(__assert_perror_fail+0x0)[0xffffbcd8e694]
[st-master:28398] [ 5] /users/hpritchard/ompi/install_master_ucx/lib/libopen-pal.so.0(mca_base_var_group_deregister+0x1f4)[0xffffbcc74378]
[st-master:28398] [ 6] /users/hpritchard/ompi/install_master_ucx/lib/libopen-pal.so.0(mca_base_var_group_deregister+0x348)[0xffffbcc744cc]
[st-master:28398] [ 7] /users/hpritchard/ompi/install_master_ucx/lib/libopen-pal.so.0(mca_base_framework_close+0x114)[0xffffbcc76858]
[st-master:28398] [ 8] /users/hpritchard/ompi/install_master_ucx/lib/libopen-pal.so.0(opal_info_close_components+0xbc)[0xffffbcc27448]
[st-master:28398] [ 9] /users/hpritchard/ompi/install_master_ucx/lib/libmpi.so.0(ompi_info_close_components+0xb0)[0xffffbcff3b4c]
[st-master:28398] [10] /users/hpritchard/ompi/install_master_ucx/lib/libmpi.so.0(PMPI_T_finalize+0x4c)[0xffffbd0a0c08]
[st-master:28398] [11] ./init_after_finalize[0x401174]
```
when compiling master with the following config options:

```
./configure --with-ucx=path_to_ucx_install_v1.10.0  --with-ofi=no --enable-debug
./configure --with-ucx=path_to_ucx_install_v1.10.0  --with-ofi=no --enable-debug --enable-mca-dso
./configure --with-ofi=path_to_ofi_install_1.12.1 --with-ucx=no --enable-debug
./configure --with-ofi=path_to_ofi_install_1.12.1 --with-ucx=no --enable-debug --enable-mca-dso
```
test works with the v4.1.x branch.  presumably works on v4.0.x too.

**Comment by hppritcha** [comment]
interestingly the test works with the sessions prototype, which is currently rebased off of master at sha c2e3fa02d0bd472e82b1f0223f33aa8fe1467616.

**Comment by gpaulsen** [comment]
Hope to get this fixed for v5.0.0 after merging https://github.com/open-mpi/ompi/pull/9097
